### ISSUE TITLE
Backspace not working, gives an error

### ISSUE THREAD

**Issue by sschuldenzucker** [syl20bnr/spacemacs#3949]
Typing backspace in insert mode results in the following error (and no effect
except for that):

    apply: Symbol's value as variable is void: smartparens-strict-mode

#### Workaround

Activate and de-activate again `smartparens-globally` via `SPC
t C-p` twice.
Thanks to @wginolas for pointing this out.

#### Misc Info

- The issue did not exist in v0.104.4, but is new in v0.104.5.
- The issue also appears with an empty `.spacemacs`.
- This appeared first as a comment to #690.

#### System Info
- OS: gnu/linux
- Emacs: 24.4.1
- Spacemacs: 0.104.5
- Spacemacs branch: master (rev. c0de218)
- Distribution: spacemacs
- Layers:
```elisp
(emacs-lisp markdown org latex spell-checking auto-completion syntax-checking version-control)
```

**Comment by Arbitrageur** [comment]
Have the same problem.

And reloading config file by pressing 'SPC f e R' as another workaround.

Maybe the order of loading matters.

**Comment by StreakyCobra** [comment]
@Arbitrageur @sschuldenzucker Have you tried updating the packages?

**Comment by sschuldenzucker** [comment]
@StreakyCobra : You mean, by deleting `.emacs.d/elpa/` and restarting?

**Comment by StreakyCobra** [comment]
Nope, by clicking on `[Update]` on the spacemacs buffer :-)

**Comment by Arbitrageur** [comment]
@StreakyCobra Yes.. but not working

**Comment by StreakyCobra** [comment]
Ok, can you paste it the backtrace? (`SPC t D` before doing the problematic action)

**Comment by Arbitrageur** [comment]
Debugger entered--Lisp error: (void-variable smartparens-strict-mode)
  ad-Advice-evil-delete-backward-char-and-join(#[(count) "\204n\204\f\301\302!\207\303\304!\207" [evil-backspace-join-lines call-interactively delete-backward-char user-error "Beginning of line"] 2 ("/Users/ykwon/.emacs.d/elpa/evil-20151109.829/evil-commands.elc" . 107921) (list (prefix-numeric-value current-prefix-arg))] 1)
  apply(ad-Advice-evil-delete-backward-char-and-join #[(count) "\204n\204\f\301\302!\207\303\304!\207" [evil-backspace-join-lines call-interactively delete-backward-char user-error "Beginning of line"] 2 ("/Users/ykwon/.emacs.d/elpa/evil-20151109.829/evil-commands.elc" . 107921) (list (prefix-numeric-value current-prefix-arg))] 1)
  evil-delete-backward-char-and-join(1)
  call-interactively(evil-delete-backward-char-and-join nil nil)
  command-execute(evil-delete-backward-char-and-join)

**Comment by TheBB** [comment]
That's weird. The advice defines that variable.

**Comment by StreakyCobra** [comment]
Steps to reproduce:

- Open Spacemacs 0.104.5 with a fresh `.spacemacs` and up-to-date packages
- Press `q` to arrive on `*scratch*` buffer
- Enter insert mode with `i`
- Enter `(` and, still in insert mode, try to delete it with `BACKSPACE`

**Comment by TheBB** [comment]
:point_right: #3959

**Comment by syl20bnr** [comment]
Fixed by @TheBB and released in 0.104.6 as well.
